client.go

```diff
@@ -315,14 +318,28 @@ func (a *Client) CopyFromRemotePassThru(
 	remotePath string,
 	passThru PassThru,
 ) error {
+	_, err := a.CopyFromRemoteFileInfos(ctx, w, remotePath, passThru)
+
+	return err
+}
+
+// CopyFroRemoteFileInfos copies a file from the remote to a given writer and return a FileInfos struct
+// containing information about the file such as permissions, the file size, modification time and access time
+func (a *Client) CopyFromRemoteFileInfos(
```

Comment:
Perhaps split this off into a private function that accepts a boolean to turn the `preserve` flag on or off. This way the public facing function will always keep track of `fileInfos` but you will avoid code duplication.

Reply:
created a private method to receive the information as a parameter instead of using the configurer.

Reply:
This is a minor one, but I don't like the suffix "functionality" after `copyFromRemote`, why not just `copyFromRemote`? I realize that there already is an existing public method with the same name, but Go's methods are case-sensitive right?

Reply:
I agree. That's what I wanted to name it initially, but decided to add an extra word just to avoid confusion when glancing at the code for the first time. I will change it to copyFromRemote